eos-tech-support/eos-download-image

```diff
@@ -173,12 +177,30 @@ class EosDownloadImage(object):
 
         return keyring
 
-    def download_and_verify(self, meta, what):
+    def download_and_verify(self, meta, what, url=None, size=None):
+        """ Download and verify image. Caller should either use meta or specify url and size."""
         destdir = self.args.outdir
-        image_url = '{}/{}'.format(self.base, meta['file'])
-        image_size = meta['compressed_size']
-        image_mtime = dateutil.parser.parse(meta['last_modified'])
-        signature_url = '{}/{}'.format(self.base, meta['signature'])
+        if meta is not None:
+            image_url = '{}/{}'.format(self.base, meta['file'])
+            image_size = meta['compressed_size']
+            image_mtime = dateutil.parser.parse(meta['last_modified'])
+            signature_url = '{}/{}'.format(self.base, meta['signature'])
+            if meta.get('extracted_signature'):
+                extracted_sig_url = '{}/{}'.format(self.base, meta['extracted_signature'])
+            else:
+                extracted_sig_url = None
+        else:
+            assert(url is not None and size is not None)
+            image_url = url
+            image_size = size
+            image_mtime = None
+            signature_url = url + '.asc'
+            if '.xz' in signature_url:
+                extracted_sig_url = signature_url.replace('.xz', '')
```

Comment:
Since you're not downloading the `boot.zip` and `boot.zip.asc` files, there's no need to download the uncompressed signature. It is only used on live USBs created with `eos-write-live-image`, which requires the `boot.zip`.